lib/fathead/flask/parse.py

```diff
@@ -0,0 +1,178 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+import os
+from bs4 import BeautifulSoup
+
+URL_ROOT = 'http://flask.pocoo.org/docs/0.11/'
+DOWNLOADED_HTML_PATH = 'download/'
+
+class Data(object):
+    """
+    Class to store parsed data
+    """
+    
+    def __init__(self, file):
+        self.HTML = ""
+        self.FILE = file
+        self.load_data()
+
+    def load_data(self):
+        with open(self.FILE,'r') as data_file:
+            document = data_file.read()
+            self.HTML = document
+
+    def get_raw_data(self):
+        return self.HTML
+
+    def get_file(self):
+        return self.FILE
+
+
+class Parser(object):
+    """
+    Parser to parse HTML files
+    """
+    
+    def __init__(self, data_object):
+        self.file_being_used = data_object.get_file()
+        self.file_name = self.file_being_used.split('/').pop()
+        self.sections = []
+        self.soup = BeautifulSoup(data_object.get_raw_data(), 'html.parser')
+
+    def parse_for_data(self):
+        for header in self.soup.find_all(['h2']):
+            section = self.parse_section(header)
+            section['example'] = self.clean_formatting(self.get_example(header))
+            if section['paragraph'] and section['example']:
+                section['abstract'] = '{}<br><pre><code>{}</code></pre>'.format(section['paragraph'], section['example'])
+                self.sections.append(section)
+            elif section['paragraph'] and not section['example']:
+                section['abstract'] = '{}'.format(section['paragraph'])
```

Comment:
Please ensure paragraph text (anything that isn't code inside the abstract) is wrapped in `<p>`